File: src/main/java/be/spring/app/service/DoodleService.java
package be.spring.app.service;

import be.spring.app.model.Account;
import be.spring.app.model.Match;
import be.spring.app.model.Presence;

import java.util.Set;

/**
 * Created by u0090265 on 10/1/14.
 */
public interface DoodleService {
    public String changeMatchPresence(Account account, long matchId, boolean present);

    Presence changePresence(Account account, long accountId, long doodleId);

    boolean sendDoodleNotificationsFor(Match match, Set<Account> accounts);
}

File: src/main/webapp/resources/angular/services/doodleService.js
'use strict';
app.factory('doodleService', function ($http) {

    return {
        getMatchDoodles: function (page) {
            return $http.get('/api/v1/matchDoodle?page=' + page + '&size=10');
        },

        getMatchDoodle: function (matchId) {
            return $http.get('/api/v1/matchDoodle/' + matchId);
        },

        changePresence: function (accountId, matchId) {
            if (accountId && matchId) {
                return $http.put('/api/v1/doodle/match/' + matchId + '/presence/' + accountId);
            }
        },
        getPresenceClass: function (presence) {
            switch (presence.type) {
                case 'NOT_FILLED_IN':
                    return "glyphicon glyphicon-question-sign grey";
                case 'PRESENT':
                    return "glyphicon glyphicon-ok green";
                case 'NOT_PRESENT':
                    return "glyphicon glyphicon-remove red";
                default:
                    return "glyphicon glyphicon-lock grey";
            }
        }
    }
});
